src/main/java/org/example/service/TableService.java

```diff
@@ -0,0 +1,43 @@
+package org.example.service;
+
+import org.example.model.Developer;
+
+import java.io.PrintWriter;
+import java.util.List;
+
+public class TableService {
+
+    private final DeveloperService developerService = DeveloperService.getInstance();
+
+    public void generateHtmlTable(PrintWriter printWriter) {
```

Comment:
It is bad practice to send a response in the service files.
The response should be sent in scope of the controllers.

Reply:
fix(but have some questions about this issue)